cmd/imagescan/collector/collector.go

```diff
@@ -110,22 +112,48 @@ func (c *Collector) Collect(ctx context.Context) error {
 		DisabledAnalyzers: disabledAnalyzers,
 	})
 	if err != nil {
-		return fmt.Errorf("creating an artifact: %w", err)
+		return fmt.Errorf("creating image artifact: %w", err)
 	}
 
 	arRef, err := artifact.Inspect(ctx)
 	if err != nil {
-		return fmt.Errorf("inspecting an artifact: %w", err)
+		return fmt.Errorf("inspecting image artifact: %w", err)
 	}
 
 	manifest, err := img.Manifest()
 	if err != nil {
 		return fmt.Errorf("extracting manifest from an artifact: %w", err)
 	}
-
+	manifestBytes, err := json.Marshal(manifest)
+	if err != nil {
+		return fmt.Errorf("marshalling manifest: %w", err)
+	}
 	digest, err := img.Digest()
 	if err != nil {
-		return fmt.Errorf("extract manifest digest: %w", err)
+		return fmt.Errorf("extracting manifest digest: %w", err)
+	}
+
+	index, err := img.IndexManifest()
+	if err != nil && !errors.Is(err, itypes.ErrImageIndexNotFound) {
+		return fmt.Errorf("extracting index manifest: %w", err)
+	}
+	indexBytes, err := json.Marshal(index)
```

Comment:
index can be nil here. In such case would be safer to skip marshal and  calling img.IndexDigest()